tests/Exception/ReadOnlyExceptionTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace tests\TomPHP\ConfigServiceProvider\Exception;
+
+use PHPUnit_Framework_TestCase;
+use TomPHP\ConfigServiceProvider\Exception\ReadOnlyException;
+
+final class ReadOnlyExceptionTest extends PHPUnit_Framework_TestCase
+{
+    public function testItImplementsTheBaseExceptionType()
+    {
+        $this->assertInstanceOf(
+            'TomPHP\ConfigServiceProvider\Exception\Exception',
+            new ReadOnlyException()
+        );
+    }
```

Comment:
@tomphp 

Not sure, maybe worth extracting an `AbstractExceptionTestCase`?

```php
class AbstractExceptionTestCase
{
    /**
     * @return string
     */
    abstract protected function className();

    final public function testItImplementsTheBaseExceptionType()
    {
        $exception = new $this->className();

        $this->assertInstanceOf('TomPHP\ConfigServiceProvider\Exception\Exception', $exception);
    }
}
```

But maybe overkill, hehe.